tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -66,37 +59,50 @@ public ExporterSettings()
+            // All the logic to assign the transport will be in the ImmutableExporterSettings .ctor
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ExporterSettings"/> class.
-        /// Direct use in tests only.
-        /// </summary>
-        internal ExporterSettings(IConfigurationSource source, Func<string, bool> fileExists)
-        {
-            _fileExists = fileExists;
+            var traceAgentUrl = source?.GetString(ConfigurationKeys.AgentUri);
+            if (!string.IsNullOrWhiteSpace(traceAgentUrl))
+            {
+                if (!Uri.TryCreate(traceAgentUrl, UriKind.RelativeOrAbsolute, out var uri))
+                {
+                    ValidationWarnings.Add($"The provided Uri: ${traceAgentUrl} provided in '{ConfigurationKeys.AgentUri}' is not valid. It won't be taken into account to send traces.");
+                }
+                else
+                {
+                    AgentUri = uri;
+                }
+            }
 
-            ConfigureTraceTransport(source, out var shouldUseUdpForMetrics);
-            ConfigureMetricsTransport(source, shouldUseUdpForMetrics);
+            TracesUnixDomainSocketPath = source?.GetString(ConfigurationKeys.TracesUnixDomainSocketPath);
+            TracesPipeName = source?.GetString(ConfigurationKeys.TracesPipeName);
+            TracesPipeTimeoutMs = source?.GetInt32(ConfigurationKeys.TracesPipeTimeoutMs) ?? 0;
+            AgentHost = source?.GetString(ConfigurationKeys.AgentHost) ??
+                        // backwards compatibility for names used in the past
+                        source?.GetString("DD_TRACE_AGENT_HOSTNAME") ??
+                        source?.GetString("DATADOG_TRACE_AGENT_HOSTNAME");
 
-            PartialFlushEnabled = source?.GetBool(ConfigurationKeys.PartialFlushEnabled)
-                // default value
-                ?? false;
+            AgentPort = source?.GetInt32(ConfigurationKeys.AgentPort) ??
+                        // backwards compatibility for names used in the past
+                        source?.GetInt32("DATADOG_TRACE_AGENT_PORT");
 
-            var partialFlushMinSpans = source?.GetInt32(ConfigurationKeys.PartialFlushMinSpans);
+            DogStatsdPort = source?.GetInt32(ConfigurationKeys.DogStatsdPort) ?? 0;
+            MetricsPipeName = source?.GetString(ConfigurationKeys.MetricsPipeName);
+            MetricsUnixDomainSocketPath = source?.GetString(ConfigurationKeys.MetricsUnixDomainSocketPath);
 
-            if ((partialFlushMinSpans ?? 0) <= 0)
+            PartialFlushEnabled = source?.GetBool(ConfigurationKeys.PartialFlushEnabled) ?? false;
+            var partialFlushMinSpans = source?.GetInt32(ConfigurationKeys.PartialFlushMinSpans);
+            if (partialFlushMinSpans > 0)
             {
-                partialFlushMinSpans = 500;
+                PartialFlushMinSpans = partialFlushMinSpans.Value;
```

Comment:
Looks like the default of 500 was lost?

Reply:
default values are set in ImmutableExporterSetting now ie if it is set to 0 there, we default to 500